cloud/scope/powervs_cluster_test.go

```diff
@@ -1704,3 +1704,515 @@ func TestCreateServiceInstance(t *testing.T) {
+			},
+			expectedName: ptr.To("ClusterName-cosinstance"),
+		},
+		{
+			name:         "Resource type is cos instance and cos instance is not nil",
+			resourceType: infrav1beta2.ResourceTypeCOSInstance,
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{CosInstance: &infrav1beta2.CosInstance{Name: "CosInstanceName"}}},
+			},
+			expectedName: ptr.To("CosInstanceName"),
+		},
+		{
+			name:         "Resource type is cos bucket and cos bucket is nil",
+			resourceType: infrav1beta2.ResourceTypeCOSBucket,
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+			},
+			expectedName: ptr.To("ClusterName-cosbucket"),
+		},
+		{
+			name:         "Resource type is cos bucket and cos bucket is not nil",
+			resourceType: infrav1beta2.ResourceTypeCOSBucket,
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{CosInstance: &infrav1beta2.CosInstance{BucketName: "CosBucketName"}}},
+			},
+			expectedName: ptr.To("CosBucketName"),
+		},
+		{
+			name:         "Resource type is subnet",
+			resourceType: infrav1beta2.ResourceTypeSubnet,
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+			},
+			expectedName: ptr.To("ClusterName-vpcsubnet"),
+		},
+		{
+			name:         "Resource type is load balancer",
+			resourceType: infrav1beta2.ResourceTypeLoadBalancer,
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+			},
+			expectedName: ptr.To("ClusterName-loadbalancer"),
+		},
+		{
+			name: "Resource type is invalid",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			expectedName: nil,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		t.Run(tc.name, func(_ *testing.T) {
+			rgID := tc.clusterScope.GetServiceName(tc.resourceType)
+			g.Expect(rgID).To(Equal(tc.expectedName))
+		})
+	}
+}
+
+func TestGetVPCByName(t *testing.T) {
+	var (
+		mockVPC  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When GetVPCByName returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient:      mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"}},
+		}
+		mockVPC.EXPECT().GetVPCByName(gomock.Any()).Return(nil, fmt.Errorf("GetVPCByName returns error"))
+		output, err := clusterScope.getVPCByName()
```

Comment:
lets avoid using generic term output wherever applicable and use specific term related return values like vpcResponse ..

Reply:
Sure, renamed it to proper terms everywhere